uploader.go

```diff
@@ -184,22 +192,47 @@ func (u UploadOptions) contextWithTimeout(ctx context.Context) (context.Context,
 	return context.WithTimeout(ctx, u.BatchWindow)
 }
 
+func addToPendingBatch(u UploadOptions, inputData InputData) Batch {
+	blob := inputData.GetBlob()
+	objectType := inputData.GetObjectType()
+	b, ok := pendingBatches[objectType]
```

Comment:
What's your thinking around grouping by `objectType` vs generated key?

Reply:
The generated key is created with the first input for a batch, and is unknown to subsequent inputs for that batch.